// @mui/codemod/node/v5.0.0/jss-to-styled.test/sixth.actual.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _clsx = _interopRequireDefault(require("clsx"));

var _IconButton = _interopRequireDefault(require("@material-ui/core/IconButton"));

var _Badge = _interopRequireDefault(require("@material-ui/core/Badge"));

var _reactRedux = require("react-redux");

var _withStyles = _interopRequireDefault(require("@material-ui/styles/withStyles"));

var _ShoppingCartOutlined = _interopRequireDefault(require("@material-ui/icons/ShoppingCartOutlined"));

var recompose = _interopRequireWildcard(require("recompose"));

var _Typography = _interopRequireDefault(require("@material-ui/core/Typography"));

var _Link = _interopRequireDefault(require("modules/components/Link"));

var _jsxRuntime = require("react/jsx-runtime");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

const styles = theme => ({
  root: {
    padding: 8,
    '&&&': {
      marginLeft: 0
    }
  },
  icon: {
    fontSize: 24
  },
  badge: {
    right: 2,
    top: -8,
    backgroundColor: theme.palette.primary.main,
    color: theme.palette.common.white
  },
  transparentBadge: {
    backgroundColor: theme.palette.common.white,
    color: theme.palette.primary.main
  }
}); // eslint-disable-next-line react/display-name


const LinkToCart = /*#__PURE__*/_react.default.forwardRef((linkProps, ref) => /*#__PURE__*/(0, _jsxRuntime.jsx)(_Link.default, (0, _extends2.default)({}, linkProps, {
  to: "/order-cart/",
  ref: ref
})));

function AppAppBarCart(props) {
  const {
    classes,
    className,
    variant,
    cart,
    transparent
  } = props;
  const count = cart.count || 0;
  return variant === 'text' ? /*#__PURE__*/(0, _jsxRuntime.jsxs)(_Typography.default, {
    component: LinkToCart,
    className: className,
    children: ['Cart', " ", count > 0 && `(${count})`]
  }) : /*#__PURE__*/(0, _jsxRuntime.jsxs)(_IconButton.default, {
    color: transparent ? 'inherit' : undefined,
    component: LinkToCart,
    className: (0, _clsx.default)(classes.root, className),
    children: [/*#__PURE__*/(0, _jsxRuntime.jsx)(_ShoppingCartOutlined.default, {
      className: classes.icon
    }), count > 0 && /*#__PURE__*/(0, _jsxRuntime.jsx)(_Badge.default, {
      color: "primary",
      classes: {
        badge: (0, _clsx.default)(classes.badge, transparent && classes.transparentBadge)
      },
      badgeContent: count > 9 ? '+9' : count,
      children: ''
    })]
  });
}

AppAppBarCart.displayName = 'AppAppBarCart';
process.env.NODE_ENV !== "production" ? AppAppBarCart.propTypes = {
  classes: _propTypes.default.object.isRequired,
  className: _propTypes.default.string,
  variant: _propTypes.default.oneOf(['text', 'icon']),
  transparent: _propTypes.default.bool
} : void 0;
AppAppBarCart.defaultProps = {
  variant: 'icon'
};

var _default = recompose.compose((0, _withStyles.default)(styles, {
  name: 'AppAppBarCart'
}), (0, _reactRedux.connect)(state => ({
  cart: state.data.cart
})))(AppAppBarCart);

exports.default = _default;
