// @mui/codemod/node/v5.0.0/jss-to-tss-react.test/expected-from-material-ui-core-styles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = ComponentUsingStyles;

var _react = _interopRequireDefault(require("react"));

var _mui = require("tss-react/mui");

var _jsxRuntime = require("react/jsx-runtime");

var _InnerComponent;

/*
Sandboxes for verifying correct behavior:
JSS - https://codesandbox.io/s/case1-jss-dedp2f?file=/src/App.js
TSS - https://codesandbox.io/s/case1-tss-s0z7tx?file=/src/App.js
 */
const useStyles = (0, _mui.makeStyles)({
  name: "TestName"
})({
  test: {
    backgroundColor: "purple",
    color: "white"
  }
});
const useStyles2 = (0, _mui.makeStyles)()(() => ({
  test2: {
    backgroundColor: "blue",
    color: "lime"
  }
}));

function InnerComponent() {
  const {
    classes
  } = useStyles2();
  return /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
    className: classes.test2,
    children: "Inner Test"
  });
}

function ComponentUsingStyles(props) {
  const {
    classes
  } = useStyles();
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
    className: classes.test,
    children: ["Test", _InnerComponent || (_InnerComponent = /*#__PURE__*/(0, _jsxRuntime.jsx)(InnerComponent, {}))]
  });
}
