// @mui/codemod/node/v5.0.0/jss-to-tss-react.test/expected-todo-comments.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useExportedStyles = exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _mui = require("tss-react/mui");

var _jsxRuntime = require("react/jsx-runtime");

var _InnerComponent;

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

// TODO jss-to-tss-react codemod: Unable to handle style definition reliably. ArrowFunctionExpression in CSS prop.
const useStyles = (0, _mui.makeStyles)()((_theme, _params, classes) => ({
  test: {
    backgroundColor: "purple",
    color: "white",
    [`&.${classes.qualifier}`]: {
      textDecoration: props => props.textDecoration
    }
  },
  qualifier: {}
})); // TODO jss-to-tss-react codemod: usages of this hook outside of this file will not be converted.

const useExportedStyles = (0, _mui.makeStyles)()({
  test: {
    backgroundColor: "purple",
    color: "white"
  }
}); // TODO jss-to-tss-react codemod: Unable to handle style definition reliably. Unsupported arrow function syntax.
// Arrow function has parameter type of Identifier instead of ObjectPattern (e.g. `(props) => ({...})` instead of `({color}) => ({...})`).

exports.useExportedStyles = useExportedStyles;
const useStyles2 = (0, _mui.makeStyles)()({
  test2: props => ({
    backgroundColor: "blue",
    color: "lime"
  })
}); // TODO jss-to-tss-react codemod: Unable to handle style definition reliably. Unsupported arrow function syntax.
// Arrow function has body type of BlockStatement instead of ObjectExpression.

const useStyles3 = (0, _mui.makeStyles)()({
  test3: props => {
    return {
      backgroundColor: "blue",
      color: "lime"
    };
  }
}); // TODO jss-to-tss-react codemod: '@global' is not supported by tss-react.
// See https://mui.com/material-ui/customization/how-to-customize/#4-global-css-override for alternatives.

const useGlobalStyles = (0, _mui.makeStyles)()(() => ({
  '@global': {
    '.sample': {
      backgroundColor: "purple",
      color: "white"
    }
  }
}));

function InnerComponent() {
  const {
    classes
  } = useStyles2();
  return /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
    className: classes.test2,
    children: "Inner Test"
  });
}

function ComponentUsingStyles(props) {
  const {
    classes,
    cx
  } = useStyles(props, {
    props: props
  });
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)(_jsxRuntime.Fragment, {
    children: [/*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
      className: classes.test,
      children: ["Test", _InnerComponent || (_InnerComponent = /*#__PURE__*/(0, _jsxRuntime.jsx)(InnerComponent, {}))]
    }), /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
      className: cx(classes.test, classes.qualifier),
      children: "Qualifier Test"
    })]
  });
} // TODO jss-to-tss-react codemod: usages of this hook outside of this file will not be converted.


var _default = useStyles3;
exports.default = _default;
